NitroxServer/ConsoleCommands/KickCommand.cs

```diff
@@ -14,31 +14,46 @@ internal class KickCommand : Command
         private readonly EntitySimulation entitySimulation;
         private readonly PlayerManager playerManager;
 
-        public KickCommand(PlayerManager playerManager, EntitySimulation entitySimulation) : base("kick", Perms.ADMIN, "Kicks a player from the server", true)
+        public KickCommand(PlayerManager playerManager, EntitySimulation entitySimulation) : base("kick", Perms.MODERATOR, "Kicks a player from the server")
         {
-            this.playerManager = playerManager;
-            this.entitySimulation = entitySimulation;
-
             AddParameter(new TypePlayer("name", true));
             AddParameter(new TypeString("reason", false));
+
+            AllowedArgOverflow = true;
+
+            this.playerManager = playerManager;
+            this.entitySimulation = entitySimulation;
         }
 
         protected override void Execute(CallArgs args)
         {
             Player playerToKick = args.Get<Player>(0);
 
-            playerToKick.SendPacket(new PlayerKicked(args.GetTillEnd(1)));
-            playerManager.PlayerDisconnected(playerToKick.connection);
-
-            List<SimulatedEntity> revokedEntities = entitySimulation.CalculateSimulationChangesFromPlayerDisconnect(playerToKick);
-            if (revokedEntities.Count > 0)
+            if (args.SenderName == playerToKick.Name)
             {
-                SimulationOwnershipChange ownershipChange = new SimulationOwnershipChange(revokedEntities);
-                playerManager.SendPacketToAllPlayers(ownershipChange);
+                SendMessage(args.Sender, "You can't kick yourself");
+                return;
             }
 
-            playerManager.SendPacketToOtherPlayers(new Disconnect(playerToKick.Id), playerToKick);
-            SendMessage(args.Sender, $"The player {playerToKick.Name} has been disconnected");
+            if (args.IsConsole || playerToKick.Permissions < args.Sender.Value.Permissions)
+            {
+                playerToKick.SendPacket(new PlayerKicked(args.GetTillEnd(1)));
+                playerManager.PlayerDisconnected(playerToKick.connection);
+
+                List<SimulatedEntity> revokedEntities = entitySimulation.CalculateSimulationChangesFromPlayerDisconnect(playerToKick);
+                if (revokedEntities.Count > 0)
+                {
+                    SimulationOwnershipChange ownershipChange = new(revokedEntities);
+                    playerManager.SendPacketToAllPlayers(ownershipChange);
+                }
+
+                playerManager.SendPacketToOtherPlayers(new Disconnect(playerToKick.Id), playerToKick);
+                SendMessage(args.Sender, $"The player {playerToKick.Name} has been disconnected");
+            }
+            else
+            {
+                SendMessage(args.Sender, $"You're not allowed to kick {playerToKick.Name}");
+            }
```

Comment:
Minor: return early (flip the `if` for "not allowed to kick") to reduce indentation of the code here.